posts/2024-09-24-seal_of_approval-mlr3/index.qmd

```diff
@@ -0,0 +1,104 @@
+
+## Relationship with `data.table`
+
+`mlr3` was designed to integrate closely with `data.table` for efficient data handling in
+machine learning workflows. There are two main ways `mlr3` is related to `data.table`:
+
+1. **Data Backend**: `mlr3` uses `data.table` as the core data backend for all `Task` objects.
+   This means that when you work with tasks in `mlr3`, the underlying data is stored and
+   managed using `data.table`. Moreover, users can leverage `data.table` syntax directly
+   within `mlr3` workflows. Accessing task data via `task$data()` returns a `data.table`,
+   enabling you to apply `data.table` operations for data preprocessing, feature engineering,
+   and subsetting without any additional conversion or overhead.
+2. **Result Storage**: `mlr3` stores various results such as predictions, resampling outcomes,
+   and benchmarking results as `data.table` objects.
+
+## Overview
+
+*Excerpted from the [`mlr3` book](https://mlr3book.mlr-org.com/chapters/chapter1/introduction_and_overview.html)*
+
+The mlr3 universe includes a wide range of tools taking you from basic ML to complex experiments.
+To get started, here is an example of the simplest functionality – training a model and making predictions.
+
+```{r}
+#| label: setup
+#| include: false
+lgr::get_logger("mlr3")$set_threshold("warn")
+lgr::get_logger("bbotk")$set_threshold("warn")
+```
+
+```{r}
+library(mlr3)
+
+task = tsk("penguins")
+split = partition(task)
+learner = lrn("classif.rpart")
+
+learner$train(task, row_ids = split$train)
+learner$model
+```
+
+```{r}
+prediction = learner$predict(task, row_ids = split$test)
+prediction
+```
+
+```{r}
+prediction$score(msr("classif.acc"))
+```
+
+In this example, we trained a decision tree on a subset of the `penguins` dataset,
+made predictions on the rest of the data and then evaluated these with the accuracy measure.
+
+The `mlr3` interface also lets you run more complicated experiments in just a few lines of code:
+
+```{r}
+#| warning: false
+library(mlr3verse)
+
+tasks = tsks(c("german_credit", "sonar"))
+
+glrn_rf_tuned = as_learner(ppl("robustify") %>>% auto_tuner(
+    tnr("grid_search", resolution = 5),
+    lrn("classif.ranger", num.trees = to_tune(200, 500)),
+    rsmp("holdout")
+))
+glrn_rf_tuned$id = "RF"
+
+glrn_stack = as_learner(ppl("robustify") %>>% ppl("stacking",
+    lrns(c("classif.rpart", "classif.kknn")),
+    lrn("classif.log_reg")
+))
+glrn_stack$id = "Stack"
+
+learners = c(glrn_rf_tuned, glrn_stack)
+bmr = benchmark(benchmark_grid(tasks, learners, rsmp("cv", folds = 3)))
+
+bmr$aggregate(msr("classif.acc"))
+```
+
+In this (much more complex!) example we chose two tasks and two learners and used automated
+tuning to optimize the number of trees in the random forest learner, and a machine learning
+pipeline that imputes missing data, collapses factor levels, and stacks models.
+We also showed basic features like loading learnersand choosing resampling strategies for benchmarking.
```

Comment:
learners and

Reply:
Also the para above could do with slight changes or subtle rephrasing, for e.g.:

> In this (much more complex!) example we chose two tasks and two learners and ...

In this (much more complex!) example, we chose two tasks and two learners.  *Insert new setence to go with the verb 'used' for both 'automated tuning' and the 'machine learning pipeline'*

or if you prefer to keep it as one sentence, then maybe we can insert a verb before 'a machine learning pipeline' (similar to 'chose' and 'used'), for e.g.:

In this more complex example, we selected two tasks and two learners, used automated tuning to optimize the number of trees in the random forest learner, and employed a machine learning pipeline that imputes missing data, consolidates factor levels, and stacks models.

Reply:
The text is copied from the book, but you're right the sentence does drag a bit.